lib/usdUfe/ufe/UsdAttributeHolder.cpp

```diff
@@ -59,17 +61,58 @@ bool setUsdAttrMetadata(
 
     UsdUfe::AttributeEditRouterContext ctx(attr.GetPrim(), attr.GetName());
 
+    // These metadata have a USD equivalent. Use it to stay compatible with other DCCs:
+    static const auto knownUSDMetadata
+        = std::unordered_set<std::string> { UsdUfe::MetadataTokens->UIDoc,
+                                            UsdUfe::MetadataTokens->UIEnumLabels,
+                                            UsdUfe::MetadataTokens->UIName,
+                                            UsdUfe::MetadataTokens->UIFolder };
+
     PXR_NS::TfToken tok(key);
     if (PXR_NS::UsdShadeNodeGraph(attr.GetPrim())) {
         if (PXR_NS::UsdShadeInput::IsInput(attr)) {
             PXR_NS::UsdShadeInput input(attr);
-            input.SetSdrMetadataByKey(tok, value.get<std::string>());
-            return true;
+            if (knownUSDMetadata.count(key)) {
+                // Cleanup known USD metadata previously set as SdrMetadata
+                input.ClearSdrMetadataByKey(tok);
+            } else {
+                input.SetSdrMetadataByKey(tok, value.get<std::string>());
+                return true;
+            }
         } else if (PXR_NS::UsdShadeOutput::IsOutput(attr)) {
             PXR_NS::UsdShadeOutput output(attr);
-            output.SetSdrMetadataByKey(tok, value.get<std::string>());
-            return true;
+            if (knownUSDMetadata.count(key)) {
+                // Cleanup known USD metadata previously set as SdrMetadata
+                output.ClearSdrMetadataByKey(tok);
+            } else {
+                output.SetSdrMetadataByKey(tok, value.get<std::string>());
+                return true;
+            }
+        }
+    }
+
+    if (key == UsdUfe::MetadataTokens->UIDoc) {
```

Comment:
[Readability] Would it make sense to factor out this code into a separate function? Something like:
```C++
namespace {
bool setUsdMetadata(UsdAttribute attribute, std::string key, Ufe::Value value)
{
    if (key == UsdUfe::MetadataTokens->UIDoc) {
        attr.SetDocumentation(value.get<std::string>());
        return true;
    } else if (key == UsdUfe::MetadataTokens->UIEnumLabels) {
    // ...
}
}
```

In the code above, we could then do something like this:
```C++
        } else if (PXR_NS::UsdShadeOutput::IsOutput(attr)) {
            if (knownUSDMetadata.count(key)) {
                // Cleanup known USD metadata previously set as SdrMetadata
                output.ClearSdrMetadataByKey(tok);
                // Set the USD metadata
                return setUsdMetadata(attr, key, value);
            }

            output.SetSdrMetadataByKey(tok, value.get<std::string>());
            return true;
        }
```

This way, we could get rid of the else block to reduce indentation by a bit and the clearing of the Sdr metadata and the setting of the USD metadata would be right next to each other, which might improve readability.

Reply:
Done. Also added missing hasMetadata and clearMetadata callbacks as well as a full featured unit test making sure UFE and USD are in sync.